src/extension/app/store/app.js

```diff
@@ -916,32 +922,28 @@ export class AppStore {
         }, { once: true });
         this.fetchStatus();
       } else if (status.webPath.startsWith('/.helix/') && resp.error) {
-        this.showModal({
-          type: MODALS.ERROR,
-          data: {
-            message: `${this.i18n('error_config_failure')}${resp.error}`,
-          },
-        });
+        this.showToast(`${this.i18n('error_config_failure')}${resp.error}`, 'negative');
       } else {
         // eslint-disable-next-line no-console
         console.error(resp);
-        this.showModal({
-          type: MODALS.ERROR,
-          data: {
-            message: this.i18n('error_preview_failure'),
-          },
-        });
+        this.showToast(`${this.i18n('error_preview_failure')}${resp.error}`, 'negative');
       }
       return;
     }
     // handle special case /.helix/*
     if (status.webPath.startsWith('/.helix/')) {
       this.showToast(this.i18n('preview_config_success'), 'positive');
-      this.hideWait();
+      this.setState();
       return;
     }
-    this.hideWait();
-    this.switchEnv('preview');
+
+    /* istanbul ignore next 4 */
+    const actionCallback = () => {
+      this.setState();
+      this.switchEnv('preview');
+    };
+
+    this.showToast(this.i18n('preview_success'), 'positive', undefined, actionCallback, 'Open');
```

Comment:
`Open` should be translated.

Should the toast handling be moved to the preview plugin like we do for other plugins?